### PR TITLE
Update test CMakeLists to add new tests automatically and add Jenkins stage for tests

### PR DESCRIPTION
**Pull request by JehandadKhan**
This PR depends on #86 

The test needs to be a single cpp file with its own main function.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -30,26 +45,44 @@ message("OpenMP_gomp_LIBRARY: ${OpenMP_gomp_LIBRARY}")
...
+    message(STATUS "CK_hip_VERSION_MAJOR overriden with ${MIOPEN_OVERRIDE_HIP_VERSION_MAJOR}")
+endif()
+if( DEFINED CK_OVERRIDE_HIP_VERSION_MINOR )
+    set(CK_hip_VERSION_MINOR "${MIOPEN_OVERRIDE_HIP_VERSION_MINOR}")
+    message(STATUS "CK_hip_VERSION_MINOR overriden with ${MIOPEN_OVERRIDE_HIP_VERSION_MINOR}")
+endif()
+if( DEFINED CK_OVERRIDE_HIP_VERSION_PATCH )
+    set(CK_hip_VERSION_PATCH "${MIOPEN_OVERRIDE_HIP_VERSION_PATCH}")
+    message(STATUS "CK_hip_VERSION_PATCH overriden with ${MIOPEN_OVERRIDE_HIP_VERSION_PATCH}")
+endif()
```
Comment (JehandadKhan) [inline CMakeLists.txt:71]: I will remove these definitions since they are not required.

File: composable_kernel/include/hip_version.hpp.in [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#pragma once
```
Comment (JehandadKhan) [inline composable_kernel/include/hip_version.hpp.in:1]: This file will be removed.

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+half,https://github.com/pfultz2/half/archive/1.12.0.tar.gz -X header -H sha256:0a08660b68abb176ebc2a0cdf8de46e3182a7f46c66443bb80dbfaaec98cf969 --build
```
Comment (JehandadKhan) [inline requirements.txt:1]: This will be removed since half is included in the repo

**Comment by JehandadKhan** [comment]
@asroy  Please note that the newly added test fails when run on certain machines. You can reproduce the failure reliably using the following commands

``` bash
 ssh fpadmin@10.236.12.127

 docker run -it -u 1002:1002 --device=/dev/kfd --device=/dev/dri --group-add video --cap-add=SYS_PTRACE --security-opt seccomp=unconfined -v=/var/jenkins/:/var/jenkins -w /var/jenkins/workspace/_composable_kernel_jd_ctest_init -v /var/jenkins/workspace/_composable_kernel_jd_ctest_init:/var/jenkins/workspace/_composable_kernel_jd_ctest_init:rw,z -v /var/jenkins/workspace/_composable_kernel_jd_ctest_init@tmp:/var/jenkins/workspace/_composable_kernel_jd_ctest_init@tmp:rw,z 2b1d0ee1b225 bash

 cd build/
 ./bin/test_conv2d_fwd
in_n_c_hi_wi: dim 4, lengths {128, 192, 71, 71}, strides {967872, 1, 13632, 192}
wei_k_c_y_x: dim 4, lengths {256, 192, 3, 3}, strides {1728, 1, 576, 192}
out_n_k_ho_wo: dim 4, lengths {128, 256, 36, 36}, strides {331776, 1, 9216, 256}
arg.a_grid_desc_k0_m_k1_{432, 165888, 4}
arg.b_grid_desc_k0_n_k1_{432, 256, 4}
arg.c_grid_desc_m_n_{ 165888, 256}
launch_and_time_kernel: grid_dim {1296, 1, 1}, block_dim {256, 1, 1}
Warm up
Start running 0 times...
test conv2d fwd: Fail
```

**Review by asroy — APPROVED** [review]

**Comment by asroy** [comment]
I cannot find reproduce the test_conv2d_fwd failure on ROCm4.3. Maybe this is a compiler issue.

Can we change the CI docker to ROCm4.3?

**Comment by asroy** [comment]
Jekins failed:
http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2Fcomposable_kernel/detail/jd%2Fctest_init/11/pipeline/#step-62-log-86
image

**Comment by JehandadKhan** [comment]
@asroy CI passes

**Review by asroy — APPROVED** [review]

**Pull request merged**